Sources/NIOHTTP2/HTTP2CommonInboundStreamMultiplexer.swift

```diff
@@ -451,3 +451,123 @@ internal protocol AnyContinuation {
     func finish()
     func finish(throwing error: Error)
 }
+
+
+/// `NIOHTTP2InboundStreamChannels` provides access to inbound stream channels as a generic `AsyncSequence`.
+/// They make use of generics to allow for wrapping the stream `Channel`s, for example as `NIOAsyncChannel`s or protocol negotiation objects.
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+public struct NIOHTTP2InboundStreamChannels<Output>: AsyncSequence {
```

Comment:
Should we nest this type inside the `AsyncStreamMultiplexer `?

Reply:
I don't have a strong opinion here but I'm not convinced. I don't know if this type _has_ to be tightly coupled to the async multiplexer and since this is public API maybe we should keep our options open?

Reply:
Fair enough!